compiler/src/iree/compiler/Dialect/Stream/IR/StreamOpFolders.cpp

```diff
@@ -2331,6 +2333,518 @@ void AsyncConcurrentOp::getCanonicalizationPatterns(RewritePatternSet &results,
   results.insert<ElideUnusedOp<AsyncConcurrentOp>>(context);
 }
 
+//===----------------------------------------------------------------------===//
+// stream.async.parameter.load
+//===----------------------------------------------------------------------===//
+
+namespace {
+
+// Safely traverses through timepoint.await to find timeline-unsafe producer.
+// Only succeeds if single-use chain: producer -> await -> consumer.
+// This allows folding patterns to work with legal IR that properly awaits
+// timeline resources before consuming them.
+static Value tryLookThroughAwait(Value operand) {
+  auto awaitOp = operand.getDefiningOp<IREE::Stream::TimepointAwaitOp>();
+  if (!awaitOp)
+    return operand;
+  if (!operand.hasOneUse())
+    return operand;
```

Comment:
style nit: add braces

```suggestion
  if (!awaitOp) {
    return operand;
  }
  if (!operand.hasOneUse()) {
    return operand;
  }
```